app/controllers/users.js

```diff
@@ -28,11 +29,23 @@ module.exports.index = (req, res, next) =>
     })
     .catch(next);
 
-module.exports.signIn = (req, res, next) =>
+module.exports.buy = async (req, res, next) => {
+  try {
+    const currentUser = await req.currentUser;
```

Comment:
Why are you awaiting this? Only promises should be awaited

Reply:
Emmm actually, this is a promise jajaja

Reply:
You are right, never seen a promise saved on the req object so it confused me.
It is kinda wierd, I'd have awaited the promise when assigning it to the req instead, but I don't really have an argument against doing it like this.